### DISCUSSION TITLE
how do i get groupid using fileid and group path?

### DISCUSSION THREAD

**Discussion by mayurmungi** [LiorBanai/HDF5-CSharp#148] [Q&A] [answered]
HI,
I am using hdf csharp to write and read from hdf5 file, and before appending to existing dataset or groupid i need to check if its exist and get group id along with same call? could you please help me identify how do i get the same?
i have file id and group path a/b/c and i need get the group id if its exists?

**Comment by LiorBanai** [comment, marked as answer]
Hi,
You have 2 methods:
 Hdf5.GroupExists(fileorGroupId, nameOfGroup);

for example:
```cs
      string filename = @"d:\test.h5";
      var fileId = Hdf5.OpenFile(filename, true);
      var exist=   Hdf5.GroupExists(fileId, "/root/eit/d1");
```

for groupId:
```cs
        public static long CreateOrOpenGroup(long fileOrGroupId, string groupName)
        {

            return (Hdf5Utils.ItemExists(fileOrGroupId, groupName, Hdf5ElementType.Group))
                ? H5G.open(fileOrGroupId, Hdf5Utils.NormalizedName(groupName))
                : H5G.create(fileOrGroupId, Hdf5Utils.NormalizedName(groupName));
        }
```
